src/createNavigationContainer.js

```diff
@@ -152,6 +152,13 @@ export default function createNavigationContainer<T: *>(
       this._validateProps(nextProps);
     }
 
+    componentDidUpdate() {
+      // Clear cached _nav every tick
+      if (this._nav === this.state.nav) {
+        delete this._nav;
```

Comment:
@dantman Would it not be better to set `this._nav` to `null` when you are clearing the cache? All `delete` is doing is removing the IIRC `delete` interferes with some JS engine optimizations.